### PR TITLE
Point buffer: distinguish points and point coordinates

### PR DESCRIPTION
**Pull request by Gustl22**
It was always confusing, when having `point` _size, pos, num_, etc., if there's meant the point size or the number of coordinates in an point array. So I renamed it.
`numPoints = numCoordinates / dimension`

### CODE REVIEW

**Comment by devemux86** [comment]
Better not start refactoring variables (can have docs for that) or else repository history will be a mess and will be difficult to monitor the changes that matter in the future.

Much more important are library's performance and stability or else we'll never progress with new versions.

**Comment by Gustl22** [comment]
Why would it be a mess? 
Docs cannot replace the whole code readability and wouldn't solve the equivalent naming of different meanings.
Refactoring simplifies uncompehensable parts of library, which is much more important than the repository history(?). Its a library internal used class, so I don't see disadvantages here. Further every new feature has to use this unprecise naming convention. And refactoring doesn't influence performance.

Changes shouldn't be excluded by importance, if they're reasonable. Wasn't meant to include this change in 11.0.
Beyond all question, performance and stability are important. But it sounds like no changes can ever happen. 
Or open a dev branch, where can test unstable states and only merge them, if they seem valid.

So close here without discussion?

**Comment by devemux86** [comment]
Cannot see any improvement in naming, the existing one seems fine all these years. If cannot understand it can improve the documentation at the specific class.

**Comment by Gustl22** [comment]
So in one class `point num` means number of points. And in another it means 2 * number of points. That seems _fine_?

**Comment by devemux86** [comment]
> _fine_?

Everyone please be polite when participating in open source projects.
Several developers worked years here and it's not very nice to mock their work in any way.

Frankly I don't remember discussing e.g. in forum, that a class variable is named incorrectly.
Certainly changes for the sake of changes are not to be made.
Should start first with explicit examples to prove something is wrong, then can think how to proceed with the less intrusive way.

**Comment by Gustl22** [comment]
> Several developers worked years here and it's not very nice to mock their work in any way.

Mocking wouldn't help anyone, and wasn't meant as this. Trying to improve, not to slam things. I only not ok, when concerns are treated too fast, without investigate the reason. 

> Frankly I don't remember discussing e.g. in forum, that a class variable is named incorrectly.

I'm kind in a predicament: GitHub is an excellent place to discuss, review and show changes that belong the library itself. So it's exaggerated to spam the forum with requests of code changes, in most cases the community only is interested in the results. At least it would end in an PR in GitHub.
So should I always make a new topic in forum, then create an issue, and implement issue in a PR?

> Certainly changes for the sake of changes are not to be made.

How is this meant?

> to prove something is wrong

So I [did](https://github.com/mapsforge/vtm/pull/700#issuecomment-474254148), not in code but in semantic (?)

**Comment by devemux86** [comment]
Certainly it's not convenient for any administrator to receive a ready PR and then waste time to try and understand why is the need of the changes.

The proper order is always to discuss first and change code afterwards.
And GitHub issues list is to report a proved problem or monitor an accepted for implementation feature. It's not for any kind of premature discussion that could lead to zero changes.

Several cases and much friction can be avoided if there are explicit examples (not with words - but with class links and numbered lines) to understand if there is an issue to proceed.

**Comment by Gustl22** [comment]
Ok agreed, thanks. So when I think, I have a proved issue, I'll open one before the PR.

File: vtm-extras/src/org/oscim/tiling/source/geojson/GeoJsonTileDecoder.java [hunk 1]
```diff
@@ -268,11 +268,6 @@ private void parsePolygon(JsonParser jp)
         }
     }
 
-    private void removeLastPoint() {
-        mMapElement.pointNextPos -= 2;
-        mMapElement.index[mMapElement.indexCurrentPos] -= 2;
-    }
-
```
Comment (Gustl22) [inline vtm-extras/src/org/oscim/tiling/source/geojson/GeoJsonTileDecoder.java]: Would it be ok, to move this method to GeometryBuffer, as it is used 3 times with same implementation?

File: vtm/src/org/oscim/core/GeometryBuffer.java [hunk 2]
```diff
@@ -178,8 +188,30 @@ public PointF getPoint(int i) {
...
+    /**
+     * Get the number of coordinates in point array.
+     * Equal to the next position to insert point coordinates in points array.
+     *
+     * @return the number of coordinates in point array.
+     */
+    public int getNumCoords() {
+        return numCoords;
+    }
+
```
Comment (Gustl22) [inline vtm/src/org/oscim/core/GeometryBuffer.java]: Would it be ok to add `getPointSize` in addition to `pointNextPos`?

File: vtm/src/org/oscim/utils/Tessellator.java [hunk 3]
```diff
@@ -32,20 +32,20 @@
      * Special version for ExtrusionLayer to match indices with vertex positions.
      * Tessellates polygon in tris elements.
      *
-     * @param points       the {@link org.oscim.core.GeometryBuffer#points}
-     * @param ppos         the {@link org.oscim.core.GeometryBuffer#pointNextPos} (not needed)
-     * @param numPoints    equals ppos?
-     * @param index        the {@link org.oscim.core.GeometryBuffer#index}
-     * @param ipos         the {@link org.oscim.core.GeometryBuffer#indexCurrentPos}
```
Comment (Gustl22) [inline vtm/src/org/oscim/utils/Tessellator.java]: Would it be ok to add a PR to clarify this signature?

File: vtm-extras/src/org/oscim/tiling/source/geojson/GeoJsonTileDecoder.java (hunk 1 above)
Comment (devemux86) [reply vtm-extras/src/org/oscim/tiling/source/geojson/GeoJsonTileDecoder.java]: Isn't that specific to implementation as map element can have 2D or 3D points?

File: vtm/src/org/oscim/utils/Tessellator.java (hunk 3 above)
Comment (devemux86) [reply vtm/src/org/oscim/utils/Tessellator.java]: Sure.

File: vtm/src/org/oscim/core/GeometryBuffer.java (hunk 2 above)
Comment (devemux86) [reply vtm/src/org/oscim/core/GeometryBuffer.java]: Mean the following?
```java
public int getPointsSize() {
    return pointNextPos;
}
```

File: vtm-extras/src/org/oscim/tiling/source/geojson/GeoJsonTileDecoder.java (hunk 1 above)
Comment (Gustl22) [reply vtm-extras/src/org/oscim/tiling/source/geojson/GeoJsonTileDecoder.java]: Many methods in GeomtryBuffer are specific to implementation...or could distinguish and make if clause with type before removing. If it's no `TRIS`, it's a 2D point.

File: vtm/src/org/oscim/core/GeometryBuffer.java (hunk 2 above)
Comment (Gustl22) [reply vtm/src/org/oscim/core/GeometryBuffer.java]: Exactly.

Comment (devemux86) [reply vtm/src/org/oscim/core/GeometryBuffer.java]: Sure.

File: vtm-extras/src/org/oscim/tiling/source/geojson/GeoJsonTileDecoder.java (hunk 1 above)
Comment (devemux86) [reply vtm-extras/src/org/oscim/tiling/source/geojson/GeoJsonTileDecoder.java]: > or could distinguish and make if clause with type before removing. If it's no TRIS, it's a 2D point

To cover all options in a new method would need such check.

**Comment by Gustl22** [comment]
_See: https://groups.google.com/forum/#!topic/mapsforge-dev/KqIuEZtCTAY_

**Pull request closed without merging**